db_stress_tool/db_stress_test_base.cc

```diff
@@ -765,6 +776,20 @@ void StressTest::OperateDb(ThreadState* thread) {
         }
       }
 
+      // Assign timestamps if necessary.
+      std::string read_ts_str;
+      std::string write_ts_str;
+      Slice read_ts;
+      Slice write_ts;
+      if (ShouldAcquireMutexOnKey() && FLAGS_user_timestamp_size > 0) {
+        read_ts_str = GenerateTimestampForRead();
+        read_ts = read_ts_str;
+        read_opts.timestamp = &read_ts;
+        write_ts_str = NowNanosStr();
```

Comment:
Maybe with NowSecondsStr (or even Minutes) could cover more cases with equal timestamps. Worth to have the option for that.

Reply:
For single db instance running on a single host (as db_stress), it is relatively easy to guarantee monotonically increasing timestamps, assuming we 1) have a sufficiently accurate clock, e.g. one that returns nano secs. 2) no faulty hardware, 3) each db_stress process restarts and delays to wait out max possible clock skew, then I think we are fine here.
We already have 1), we assume 2) in this test, and 3) may be done later.